python/ray/dag/compiled_dag_node.py

```diff
@@ -582,6 +589,8 @@ def _compute(
         if self.collective_op is not None:
             # Run a NCCL collective operation.
             method = self.collective_op.execute
+        elif self.requires_nccl_read or self.requires_nccl_write:
+            method = lambda arg: arg
```

Comment:
Shouldn't the method here be something like `self.collective_op.execute`, not identity function?

Reply:
It is the identity function because:
- If the operation is a NCCL write (i.e., it corresponds to a `_NcclSendNode`), the actual work is done in `_write`. The NCCL write operation first `_read`s in the output from the actual actor task via an `IntraProcessChannel`, then it `_compute`s, which in this case, just returns the data read so that it is ready to be sent via NCCL.
- If the operation is a NCCL read, the actual work is done in `_read`. After reading in the data via NCCL, it `_compute`s (which again just returns the read data), and `_write`s the data to the actual downstream actor task via `IntraProcessChannel`.

I introduced a new interface `_SynchronousGroup` that represents a group of tasks in synchronous operations like NCCL and wrapped both the identity function here and the original `collective_op.execute` in `sync_group.execute`.

Reply:
Yes, but what I am suggesting is that we move the code that is currently in `_write` and `_read` into an `execute` method instead. So we get rid of the concept of `_write` and `_read` entirely and just have a generic `execute` method that we call instead.